streams/voxel_stream.cpp

```diff
@@ -5,6 +5,16 @@
 VoxelStream::VoxelStream() {
 }
 
+void VoxelStream::set_channel(VoxelBuffer::ChannelId channel) {
+	ERR_FAIL_INDEX(channel, VoxelBuffer::MAX_CHANNELS);
+	_channel = channel;
+	emit_changed();
+}
+
+VoxelBuffer::ChannelId VoxelStream::get_channel() const {
+	return _channel;
+}
```

Comment:
What is the reasoning behind having a `channel` property in `VoxelStream`? At this level of abstraction, a stream can provide 1, 2, or all 8 channels at once, from either procedural sources or existing files.

Reply:
I don't know? I started with it in voxel generator, but then my gdscript terrain demo wouldn't work because I couldn't inherit from voxelgenerator and get it to generate anything.

Also since you had previously said the streams (before there were generators) should dictate the channel, not the VT nodes,  I figured that every stream/generator should have the variable. 

Tbh, I don't know what the streams are for if you're going to separate the generators from them.

Reply:
I could move the functions back to the generators and ignore streams entirely.

But that generate block function passes by reference. I didn't think that could be used in gdscript? And I had other issues with gdscript generation as noted.

Reply:
`emerge_block` passed by reference too, I don't understand what the problem is

Reply:
Ok, maybe I just imagined the problem. I'll move the channel functions to voxel generator and ignore the streams. Then I'll see if I can get the gdscript working. Everything else worked but that. Maybe I just need a break from it.

Reply:
Channel functions are now only in the generators.

Creating a terrain based in Gdscript works fine. I have a [branch](https://github.com/tinmanjuggernaut/voxelgame/tree/detect_channel) in my demo that works.

Incidentally, none of your demos work with the changes you've been making recently.